### DISCUSSION TITLE
Changing Export File Names in the MATLAB wrapper for the Cuvis SDK

### DISCUSSION THREAD

**Discussion by polanco06** [cubert-hyperspectral/cuvis.sdk#57] [Q&A] [answered]
Hello everyone!

I am a student currently collaborating with a team that is leveraging the MATLAB SDK alongside the 'Ultris S5' camera by Cubert. In our workflow, we find the need to enhance categorization and search capabilities by changing the names of saved files.

As a vital component of our project, the Ultris S5 camera significantly contributes to data acquisition. However, we are encountering challenges in efficiently managing the files generated by the MATLAB SDK, particularly in terms of organization and searchability.

I would greatly appreciate any insights or suggestions from the community. Has anyone encountered a solution or have recommendations on how to modify the names of files generated by the MATLAB SDK to better facilitate categorization and search?

**Comment by robertCubert** [comment]
@BenMGeo could you please have a look?

**Comment by BenMGeo** [comment, marked as answer]
Dear @polanco06 

you can change the filenames by changing the measurements name property (`set_name`) and/or adjusting the (sub)directories of the file structure when calling `save`.

For automatic generation of the filenames, we suggest object names, dates, also parameters, if changing frequently, if this is needed for your search, separated by underscores ("_"), e.g.

A different approach is an external table where you add the information when taking a measurement, while keeping IDkeys in the filenames and the respective line in the table. 

Both approaches have pros and cons.
With the first one, you see directly what the content will be and the information is not separable from the content, while you need to search all files, when looking for something specific.
With the second one, you have a single simple table to check, while you can lose measurement or additional information while keeping the other. 

Nevertheless, you can also always readout the parameters only and generate such a table from your measurements (at least the part that is managed by the SDK automatically).

I hope that helps.

Best,
Ben
